### PR TITLE
BZ1987048: Add explanations for Advanced Storage settings

### PR DESCRIPTION
**Pull request by lmandavi**
This PR is associated with https://bugzilla.redhat.com/show_bug.cgi?id=1987048 and
https://issues.redhat.com/browse/CNV-13248?filter

### CODE REVIEW

**Comment by lmandavi** [comment]
@yaacov - Requesting SME review. Thanks!

File: modules/virt-adding-a-boot-source-web.adoc [hunk 1]
```diff
@@ -41,11 +41,15 @@ There are four methods for selecting and adding a boot source in the web console
...
-.. Advanced: Click *Storage class* and select the storage class that is used to create the disk.
+.. Optional: *Advanced Storage settings*: Click *Storage class* and select the storage class that is used to create the disk. Typically, this storage class is the default storage class that was created for use by for all PVCs.
 
-.. Advanced: Click *Access mode* and select an access mode for the persistent volume. Supported access modes are: *Single User (RWO)*, *Shared Access (RWX)*, and *Read Only (ROX)*.
+.. Optional: *Advanced Storage settings*: Click *Access mode* and select an access mode for the persistent volume. Supported access modes are:
 
-.. Advanced: Click *Volume mode* if you want to select *Block* instead of the default value *Filesystem*.
+* *Single User (RWO)* mounts the volume as read-write by a single node.
+* *Shared Access (RWX)* mounts the volume as read-only by many nodes.
+* *Read Only (ROX)* mounts the volume as read-write by many nodes.
```
Comment (yaacov) [inline modules/virt-adding-a-boot-source-web.adoc:50]: it's swapped
RWX - read-write by many
ROX - read-only by many nodes

**Comment by lmandavi** [comment]
@yaacov - Please review the updated content. Thanks!

**Comment by lmandavi** [comment]
https://deploy-preview-35415--osdocs.netlify.app/openshift-enterprise/latest/virt/vm_templates/virt-creating-vm-template.html#virt-adding-a-boot-source-web_virt-creating-vm-template

**Review by yaacov — APPROVED** [review]

**Comment by yaacov** [comment]
LGTM

**Comment by lmandavi** [comment]
@gouyang - Requesting QE review.

**Comment by lmandavi** [comment]
@gouyang - Requesting QE review. For any new documentation for a new feature, please open a new defect/Jira story. The scope of this PR is limited to the input provided in associated defect.

**Comment by lmandavi** [comment]
@yaacov - please see my notes to QE. Thank you.

**Comment by yaacov** [comment]
@lkladnit hi, can you do the QE review, since @gouyang is on PTO ?

**Comment by gouyang** [comment]
@yaacov The doc is not looking correct because we introduce new things in https://issues.redhat.com/browse/CNV-9303, I think it needs to combine the content.

**Comment by lkladnit** [comment]
Looks good.
LGTM

**Comment by yaacov** [comment]
> @yaacov The doc is not looking correct because we introduce new things in https://issues.redhat.com/browse/CNV-9303, I think it needs to combine the content.

@gouyang yes, CNV-9303 changed the way we calculate the advanced storage settings internally, but we do not document it.

AFAIK, https://issues.redhat.com/browse/CNV-9303, does not actually change the UI behavior describe in the docs, we changed the defaults to come from the new "storageProfile" CRD that is new 4.9 instead of using our configMap + some logic, the user still see a default volume and access modes as they did in older versions, the deference is how this values get calculated and AFAIU we do not document this process.

**Comment by lmandavi** [comment]
@ycui@redhat.com - @gouyang  is on PTO this week? Based on the notes I've added to this Jira story and the background explanation provided by Yaacov Zamir, could we please get a backup QE engineer to complete QE review and approval for this GitHub issue?

**Comment by gouyang** [comment]
If this is for release-4.8, it looks good to me.

For 4.9, there are two problems:
1. `Mount this as a CD-ROM boot source` is changed to `This is a CD-ROM boot source`.
2. 'VolumeMode' and 'AccessMode' isn't showing after click the "advanced storaget settings", it's hidden by the storageProfile settings.

If it's for release-4.9, can we fix problem #1 and leave #2 in https://issues.redhat.com/browse/CNV-9303.

**Comment by yaacov** [comment]
> If this is for release-4.8, it looks good to me.
> 
> For 4.9, there are two problems:
> 
>     1. `Mount this as a CD-ROM boot source` is changed to `This is a CD-ROM boot source`.
> 
>     2. 'VolumeMode' and 'AccessMode' isn't showing after click the "advanced storaget settings", it's hidden by the storageProfile settings.
> 
> 
> If it's for release-4.9, can we fix problem #1 and leave #2 in https://issues.redhat.com/browse/CNV-9303?

Thanks @gouyang I agree, in 4.9 "advanced storage settings" the 'VolumeMode' and 'AccessMode'  can be hidden, a user can hide / show this inputs by checking / unchecking "Apply optimized StorageProfile settings" checkbox to toggel between optimized values suggested by StorageProfile CRD

**Comment by lmandavi** [comment]
Requesting peer review of this GitHub issue for the CNV 4.8 release. For the two items identified by QE, I have requested QE to open DOC defects. For details, please see the Jira story:
https://issues.redhat.com/browse/CNV-13248?filter

File: modules/virt-adding-a-boot-source-web.adoc [hunk 2]
```diff
@@ -35,17 +35,21 @@ There are four methods for selecting and adding a boot source in the web console
...
+. Optional: Click *Mount this as a CD-ROM boot source* to mount a CD-ROM and use it to install the operating system on to an empty disk. The additional empty disk is automatically created and mounted by {VirtProductName}. If the additional disk is not needed, you can remove it when you create the virtual machine.
 
 . Enter a value for *Persistent Volume Claim size* to specify the PVC size that is adequate for the uncompressed image and any additional space that is required.
 
 .. Optional: Enter a name for *Source provider* to associate the name with this template.
 
-.. Advanced: Click *Storage class* and select the storage class that is used to create the disk.
+.. Optional: *Advanced Storage settings*: Click *Storage class* and select the storage class that is used to create the disk. Typically, this storage class is the default storage class that was created for use by for all PVCs.
```
Comment (sjhala-ccs) [inline modules/virt-adding-a-boot-source-web.adoc:44]: Suggestion for the last sentence: "Typically, this storage class is the default storage class that is created for use by all PVCs."

**Review by sjhala-ccs — COMMENTED** [review]
A couple of nits, otherwise LGTM

File: modules/virt-adding-a-boot-source-web.adoc [hunk 3]
```diff
@@ -35,17 +35,21 @@ There are four methods for selecting and adding a boot source in the web console
...
 
-.. Advanced: Click *Access mode* and select an access mode for the persistent volume. Supported access modes are: *Single User (RWO)*, *Shared Access (RWX)*, and *Read Only (ROX)*.
+.. Optional: *Advanced Storage settings*: Click *Access mode* and select an access mode for the persistent volume:
 
-.. Advanced: Click *Volume mode* if you want to select *Block* instead of the default value *Filesystem*.
+* *Single User (RWO)* mounts the volume as read-write by a single node.
+* *Shared Access (RWX)* mounts the volume as read-write by many nodes.
+* *Read Only (ROX)* mounts the volume as read-only by many nodes.
+
+.. Optional: *Advanced Storage settings*: Click *Volume mode* if you want to select *Block* instead of the default value *Filesystem*. {ProductName} can statically provision raw block volumes. These volumes do not have a file system, and can provide performance benefits for applications that either write to the disk directly or implement their own storage service.
```
[the remark this answers is not in the document]
Comment (lmandavi) [reply modules/virt-adding-a-boot-source-web.adoc]: Updated the content.

File: modules/virt-adding-a-boot-source-web.adoc (hunk 2 above)
Comment (lmandavi) [reply modules/virt-adding-a-boot-source-web.adoc]: Updated the content.

File: modules/virt-adding-a-boot-source-web.adoc [hunk 4]
```diff
@@ -35,17 +35,21 @@ There are four methods for selecting and adding a boot source in the web console
...
+. Optional: Click *Mount this as a CD-ROM boot source* to mount a CD-ROM and use it to install the operating system on to an empty disk. The additional empty disk is automatically created and mounted by {VirtProductName}. If the additional disk is not needed, you can remove it when you create the virtual machine.
 
 . Enter a value for *Persistent Volume Claim size* to specify the PVC size that is adequate for the uncompressed image and any additional space that is required.
 
 .. Optional: Enter a name for *Source provider* to associate the name with this template.
 
-.. Advanced: Click *Storage class* and select the storage class that is used to create the disk.
+.. Optional: *Advanced Storage settings*: Click *Storage class* and select the storage class that is used to create the disk. Typically, this storage class is the default storage class that is created for use by for all PVCs.
```
Comment (sjhala-ccs) [inline modules/virt-adding-a-boot-source-web.adoc:44]: In the last sentence, the second "for" seems redundant. Change to "..created for use by all PVCs." or "..created for all PVCs."

Comment (lmandavi) [reply modules/virt-adding-a-boot-source-web.adoc]: Updated the content.

**Pull request merged**